test/test.ads.js

```diff
@@ -325,299 +268,160 @@ QUnit.test('changing src does not trigger "contentupdate" during ad playback', f
-  assert.strictEqual(this.player.ads.state, 'content-set');
+  this.player.on('ended', endedSpy);
 
+  this.player.trigger('loadstart');
   this.player.trigger('adsready');
-  assert.strictEqual(this.player.ads.state, 'ads-ready');
-
   this.player.trigger('play');
   this.player.trigger('adtimeout');
   this.player.trigger('ended');
-  assert.strictEqual(this.player.ads.state, 'postroll?');
-
   this.player.trigger('adserror');
-  assert.strictEqual(this.player.ads.state, 'content-resuming');
-  assert.strictEqual(this.player.ads.triggerevent, 'adserror', 'adserror should be the trigger event');
 
-  this.clock.tick(1);
-  assert.strictEqual(this.player.ads.state, 'content-playback');
+  this.clock.tick(1000);
+  assert.strictEqual(endedSpy.callCount, 1, 'ended event happened');
 });
 
-QUnit.test('adtimeout in postroll? transitions to content-playback and fires ended', function(assert) {
+QUnit.test('ended event happens after postroll timed out', function(assert) {
+  var endedSpy = sinon.spy();
 
-  assert.strictEqual(this.player.ads.state, 'content-set');
+  this.player.on('ended', endedSpy);
 
+  this.player.trigger('loadstart');
   this.player.trigger('adsready');
-  assert.strictEqual(this.player.ads.state, 'ads-ready');
-
   this.player.trigger('play');
   this.player.trigger('adtimeout');
   this.player.trigger('ended');
-  assert.strictEqual(this.player.ads.state, 'postroll?');
-
   this.player.trigger('adtimeout');
-  assert.strictEqual(this.player.ads.state, 'content-resuming');
-  assert.strictEqual(this.player.ads.triggerevent, 'adtimeout', 'adtimeout should be the trigger event');
 
-  this.clock.tick(1);
-  assert.strictEqual(this.player.ads.state, 'content-playback');
+  this.clock.tick(1000);
+  assert.strictEqual(endedSpy.callCount, 1, 'ended event happened');
 });
 
-QUnit.test('adskip in postroll? transitions to content-playback and fires ended', function(assert) {
+QUnit.test('ended event happens after postroll skipped', function(assert) {
+  var endedSpy = sinon.spy();
 
-  assert.strictEqual(this.player.ads.state, 'content-set');
+  this.player.on('ended', endedSpy);
 
+  this.player.trigger('loadstart');
   this.player.trigger('adsready');
-  assert.strictEqual(this.player.ads.state, 'ads-ready');
-
   this.player.trigger('play');
-  this.player.trigger('adtimeout');
-  this.player.trigger('ended');
-  assert.strictEqual(this.player.ads.state, 'postroll?');
-
-  this.player.trigger('adskip');
-  assert.strictEqual(this.player.ads.state, 'content-resuming');
-  assert.strictEqual(this.player.ads.triggerevent, 'adskip', 'adskip should be the trigger event');
-
-  this.clock.tick(1);
-  assert.strictEqual(this.player.ads.state, 'content-playback');
+  this.player.trigger('adtimeout'); // preroll times out
+  this.player.trigger('ended'); // content ends (contentended)
+  this.player.trigger('adskip'); // postroll skipped
+  
+  this.clock.tick(1000);
```

Comment:
also here

Reply:
Yes, as long as I also update it to call `this.player.ads.skipLinearAdMode();` rather than trigger adskip.

Reply:
I guess it was timing out rather than skipping as it looked like.

Reply:
ah interesting